File: web/src/app/sharing/beer-mapper.ts
import {Beer} from '../model/Beer';
import {BeerDto} from '../model/BeerDto';

export class BeerMapper {
  public static toDto(beer: Beer) {
    let beerDto = new BeerDto();
    beerDto.name = beer.name;
    beerDto.breweryId = beer.brewery.id;
    beerDto.price = beer.price;
    beerDto.servingTemperature = beer.servingTemperature;
    beerDto.beerNoteId = beer.beerNote.id;
    beerDto.beerTypeId = beer.beerType.id;
    beerDto.alcoholPercentage = beer.alcoholPercentage;
    beerDto.ibu = beer.ibu;
    beerDto.barCode = beer.barCode;
    beerDto.id = beer.id;
    return beerDto;
  }
}

File: server/src/main/java/com/brewed/core/model/beer/BeerMapper.java
package com.brewed.core.model.beer;

import com.brewed.core.model.DtoDomainMapper;
import com.brewed.core.model.brewery.Brewery;

import static java.util.Objects.nonNull;

public class BeerMapper implements DtoDomainMapper<BeerDto, Beer> {


    @Override
    public Beer toDomain(BeerDto beerDto) {
        return Beer.builder()
                   .id(beerDto.getId())
                   .barCode(beerDto.getBarCode())
                   .alcoholPercentage(beerDto.getAlcoholPercentage())
                   .beerType(new BeerType(beerDto.getBeerTypeId()))
                   .brewery(new Brewery(beerDto.getBreweryId()))
                   .name(
                           beerDto.getName())
                   .beerNote(new BeerNote(beerDto.getBeerNoteId()))
                   .price(beerDto.getPrice())
                   .servingTemperature(beerDto.getServingTemperature())
                   .ibu(beerDto.getIbu())
                   .build();
    }

    @Override
    public BeerDto toDto(Beer beer) {
        return BeerDto.builder()
                      .alcoholPercentage(beer.getAlcoholPercentage())
                      .barCode(
                              beer.getBarCode())
                      .beerTypeId(nonNull(beer.getBeerType()) ? beer.getBeerType()
                                                                    .getId() : null)
                      .breweryId(nonNull(beer.getBrewery()) ?
                                         beer.getBrewery()
                                             .getId() : null)
                      .id(beer.getId())
                      .name(beer.getName())
                      .beerNoteId(nonNull(beer.getBeerNote()) ? beer.getBeerNote()
                                                                    .getId() : null)
                      .servingTemperature(beer.getServingTemperature())
                      .ibu(beer.getIbu())
                      .price(
                              beer.getPrice())
                      .build();
    }
}
